modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Navigation.js

```diff
@@ -15,31 +15,54 @@ import React, {useCallback, useContext, useState} from 'react';
 
 import ConnectionContext from '../context/ConnectionContext';
 import {StoreContext, useHistoricalWarning, useWarning} from '../context/store';
+import APIService from '../utils/APIService';
 import Detail from './Detail';
 import Main from './Main';
 
 export default function Navigation({
-	api,
-	authorName,
-	authorPortraitURL,
-	authorUserId,
+	author,
 	defaultTimeRange,
 	defaultTimeSpanKey,
+	endpoints,
 	languageTag,
+	namespace,
+	onSelectedLanguageClick = () => {},
+	page,
 	pagePublishDate,
 	pageTitle,
 	timeSpanOptions,
 	trafficSources,
+	viewURLs,
 }) {
+	const [{publishedToday}] = useContext(StoreContext);
+
+	const {validAnalyticsConnection} = useContext(ConnectionContext);
+
 	const [currentPage, setCurrentPage] = useState({view: 'main'});
 
 	const [trafficSourceName, setTrafficSourceName] = useState('');
 
-	const {validAnalyticsConnection} = useContext(ConnectionContext);
+	const [hasHistoricalWarning] = useHistoricalWarning();
 
 	const [hasWarning] = useWarning();
 
-	const [hasHistoricalWarning] = useHistoricalWarning();
+	const {
```

Comment:
You mean
```js
const [{historicalWarning, publishedToday, warning}] = useContext(
	StoreContext
);
```

or

```js
const [{historicalWarning}] = useHistoricalWarning();

const [{warning}] = useWarning();
```

Because I'm only using the state, not the action 🤔

Reply:
I think I'm going with 1)
What do you think?

Reply:
I was talking about 

```
const {
	getAnalyticsReportsHistoricalReadsURL,
	getAnalyticsReportsHistoricalViewsURL,
	getAnalyticsReportsTotalReadsURL,
	getAnalyticsReportsTotalViewsURL,
} = endpoints;
```

Reply:
Those are URLs, there are not objects, what do you think about to change here? 🤔

Reply:
```
export default function Navigation({
...
endpoints: {
	getAnalyticsReportsHistoricalReadsURL,
	getAnalyticsReportsHistoricalViewsURL,
	getAnalyticsReportsTotalReadsURL,
	getAnalyticsReportsTotalViewsURL,
},
...
}) {
```

Reply:
Oh, ok! 😄